Dimmer.h

```diff
@@ -1,73 +1,132 @@
+     *					Minimum is 0 and maximum is 100. Defaut is 50%.
+     * @param state 	initial state of the light. Possible states: ON or OFF. Default is OFF.
+     * @see   begin()
+     */ 
+    DimmerControl(uint8_t triacPin, uint8_t value=50, bool state = OFF);
+
+    /**
+     * Initializes the module.
+     *
+     * Initializes ZeroCross and Timer 2 interrupts. Set the light according to initial settings.
+     * @param mode 		operation mode to crontrol the light. 
+     *					Possible modes:
+     *					NORMAL_MODE: Uses timer to apply only a percentage of the AC power to the light.
+     *					RAMP_MODE: Same as in normal mode, but it applies a ramp effect to the light.
+     *					COUNT_MODE: Counts AC waves and applies full waves from time to time. @see resolution
+	 * @param resolution 	IN COUNT MODE:
+     *                          divided prescalar to define how many pulses to count.
+	 *                          if resolution is 1 the mode counts 100 pulses, if resolution is 2
+	 *                          the mode counts 50 pulses, and so on.
+     *
+     *                      IN RAMP MODE:
+	 *                          Controlls the speed of the ramp when changing values.
+     */
+    bool begin(uint8_t mode = NORMAL_MODE, uint8_t resolution = 1);
+
+    /**
+     * Turns the light OFF. 
+     */
     void off();
-    void off(byte lamp);
+
+    /**
+     * Turns the light ON. 
+     */
     void on();
-    void on(byte lamp);
-    void toggle(byte lamp);
-    void set(byte lamp, byte value);
+
+    /**
+     * Toggles the light state. 
+     */
+    void toggle();
+
+    /**
+     * Gets the current value of the dimmed light.
+     *
+     * @return current light value.
+     */
+    bool getState();
+
+    /**
+     * Gets the current state of the dimmed light.
+     *
+     * @return current light state.
+     */
+    uint8_t getValue();
+
+    /**
+     * Sets the value and the state of the light.
+     * @param value 	the value of the dimm light. Values from 0 to 100.
+     * @param state 	the state of the light. ON or OFF.
+     */
+    void set(uint8_t value, uint8_t state = -1);
+
     void zeroCross();
     void triac();
-
-private:
-    int start();
-    bool ramp_mode;
-    bool countMode;
-    byte countResolution;
-    boolean zeroCrosAttached;
-    boolean triacAttached;
-    
-    //Triac pin vector
-    byte triacPins[MAX_TRIAC];
     
-    //Triac pin counter
-    byte triacs;
-    
-    //Dimmer lamps value
-    byte lampValue[MAX_TRIAC];
-    
-    //Temporary lamps value
-    byte lampValue_ramp[MAX_TRIAC];
-    
-    //Virtual switch
-    byte lampSwitch[MAX_TRIAC];
-    
-    //50ms Counter
-    unsigned int msCounter;
-    
-    // Half cycle counter
-    byte halfCycleCounter;
+  private:
+    uint8_t operation_mode;
+    uint8_t countResolution;
+    uint8_t halfCycleCounter;
+    uint8_t triacPin;
+    uint8_t lampValue;
+    uint8_t lampValue_ramp;
```

Comment:
Let's standardize on camelcase for variables and class attributes: `lampValueRamp`.